src/installer/tests/Microsoft.NET.HostModel.Tests/AppHost.Bundle.Tests/NetCoreApp3CompatModeTests.cs

```diff
@@ -0,0 +1,91 @@
+    {
+        private SharedTestState sharedTestState;
+
+        public NetCoreApp3CompatModeTests(SharedTestState fixture)
+        {
+            sharedTestState = fixture;
+        }
+
+        [Fact]
+        public void Bundle_Is_Extracted()
+        {
+            var fixture = sharedTestState.TestFixture.Copy();
+            string singleFile;
+            Bundler bundler = BundleHelper.BundleApp(fixture, out singleFile, BundleOptions.BundleAllContent);
+            var extractionBaseDir = BundleHelper.GetExtractionRootDir(fixture);
+
+            Command.Create(singleFile, "executing_assembly_location trusted_platform_assemblies assembly_location System.Console")
+                .CaptureStdOut()
+                .CaptureStdErr()
+                .EnvironmentVariable(BundleHelper.DotnetBundleExtractBaseEnvVariable, extractionBaseDir.FullName)
+                .Execute()
+                .Should()
+                .Pass()
+                // Validate that the main assembly is running from disk (and not from bundle)
+                .And.HaveStdOutContaining("ExecutingAssembly.Location: " + extractionBaseDir.FullName)
+                // Validate that TPA contains at least one framework assembly from the extraction directory
+                .And.HaveStdOutContaining("System.Runtime.dll")
+                // Validate that framework assembly is actually loaded from the extraction directory
+                .And.HaveStdOutContaining("System.Console location: " + extractionBaseDir.FullName);
+
+            var extractionDir = BundleHelper.GetExtractionDir(fixture, bundler);
+            var bundleFiles = BundleHelper.GetBundleDir(fixture).GetFiles().Select(file => file.Name).ToArray();
+            var publishedFiles = Directory.GetFiles(BundleHelper.GetPublishPath(fixture), searchPattern: "*", searchOption: SearchOption.AllDirectories)
+                .Select(file => Path.GetFileName(file))
+                .Except(bundleFiles)
+                .ToArray();
+            var bundlerFiles = BundleHelper.GetBundleDir(fixture).GetFiles();
+            extractionDir.Should().HaveFiles(publishedFiles);
+        }
+
+        public class SharedTestState : IDisposable
+        {
+            public TestProjectFixture TestFixture { get; set; }
+            public RepoDirectoriesProvider RepoDirectories { get; set; }
+
+            public SharedTestState()
+            {
+                RepoDirectories = new RepoDirectoriesProvider();
+                TestFixture = new TestProjectFixture("SingleFileApiTests", RepoDirectories);
+                TestFixture
+                    .EnsureRestoredForRid(TestFixture.CurrentRid, RepoDirectories.CorehostPackages)
+                    .PublishProject(runtime: TestFixture.CurrentRid, outputDirectory: BundleHelper.GetPublishPath(TestFixture));
+
+                // The above will publish the app as self-contained using the stock .NET Core SDK (typically from Program Files)
```

Comment:
As far as I know dotnet/runtime switched to a mode where if the machine-wide has high enough version installed, it will use that and not download its own into .dotnet. The download only happens if the machine-wide doesn't have the right version.

Regardless, the point is actually that we get hostpolicy from non-live build (machine-wide or .dotnet, both are downloaded from official builds, not from the local machine) - and we need to test live-built binaries.